File: python/draft_templates.py
import datetime
import sdk.python.entities
from sdk.python.entities import Property
from sdk.python.files import File
from sdk.python.jobs import Job
from sdk.python.products import Product


class DraftTemplate(sdk.python.entities.Entity):

    resource = '/draft_templates/'
    json_name = 'draft_template'

    id = Property(int)
    file = Property(File)
    product = Property(Product)
    job = Property(Job)
    description = Property(str)
    name = Property(str)
    date = Property(datetime.datetime)
    height = Property(int)
    width = Property(int)
    customisation_map = Property(dict)
    customisation_map_source = Property(str)
    customisation_map_file_id = Property(str)


class DraftTemplates(sdk.python.entities.Resource):

    entity_class = DraftTemplate
    json_name = 'draft_templates'


draft_comments = DraftTemplates()

File: php/src/draft_templates.php
<?php
declare(strict_types=1);

require_once 'entity.php';
require_once 'domains.php';
require_once 'users.php';

class DraftTemplate extends Entity
{
    public static $resource = '/draft_templates/';
    public static $json_name = 'draft_template';

    public function __construct()
    {
        $this->json_property('id', 'integer');
        $this->json_property('archived', 'integer');
        $this->json_property('date', 'integer');
        $this->json_property('description', 'string');
        $this->json_property('name', 'string');
        $this->json_property('height', 'float');
        $this->json_property('width', 'float');

        $this->json_property('product', 'Product',
            $many = False, $recursive = True);
        $this->json_property('file', 'MerchiFile',
            $many = False, $recursive = True);
        $this->json_property('job', 'Job',
            $many = False, $recursive = True);
    }
}


